commit cf36c23bc952744feeb4d0b4fd81d7004bfed8cf
Author: Daniel Walsh <dwalsh@redhat.com>
Date:   Fri May 11 10:34:49 2012 +0100

    Don't enable the AppArmour security driver with LXC
    
    The AppArmour driver does not currently have support for LXC
    so ensure that when probing, it claims to be disabled
    
    Signed-off-by: Daniel P. Berrange <berrange@redhat.com>

Index: libvirt-0.9.12/src/security/security_apparmor.c
===================================================================
--- libvirt-0.9.12.orig/src/security/security_apparmor.c	2012-06-29 18:16:12.892256359 -0500
+++ libvirt-0.9.12/src/security/security_apparmor.c	2012-06-29 18:16:15.864256426 -0500
@@ -328,7 +328,7 @@
 
 /* Called on libvirtd startup to see if AppArmor is available */
 static int
-AppArmorSecurityManagerProbe(const char *virtDriver ATTRIBUTE_UNUSED)
+AppArmorSecurityManagerProbe(const char *virtDriver)
 {
     char *template = NULL;
     int rc = SECURITY_DRIVER_DISABLE;
@@ -336,6 +336,9 @@
     if (use_apparmor() < 0)
         return rc;
 
+    if (virtDriver && STREQ(virtDriver, "LXC"))
+        return rc;
+
     /* see if template file exists */
     if (virAsprintf(&template, "%s/TEMPLATE",
                                APPARMOR_DIR "/libvirt") == -1) {
